azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -113,21 +115,22 @@ def initialize(self):
                 if not self.__check_no_legacy_cgroups():
                     return
 
-                agent_unit_name = self._cgroups_api.get_agent_unit_name()
+                agent_unit_name = systemd.get_agent_unit_name()
                 agent_slice = systemd.get_unit_property(agent_unit_name, "Slice")
                 if agent_slice not in (_AZURE_SLICE, "system.slice"):
                     self.__log_cgroup_warning("The agent is within an unexpected slice: {0}", agent_slice)
                     return
 
-                self.__setup_unit_files()
+                self.__setup_azure_slice()
 
                 cpu_controller_root, memory_controller_root = self.__get_cgroup_controllers()
                 self._agent_cpu_cgroup_path, self._agent_memory_cgroup_path = self.__get_agent_cgroups(agent_slice, cpu_controller_root, memory_controller_root)
 
                 if self._agent_cpu_cgroup_path is not None:
                     self.__log_cgroup_info("Agent CPU cgroup: {0}", self._agent_cpu_cgroup_path)
-                    CGroupsTelemetry.track_cgroup(CpuCgroup(agent_unit_name, self._agent_cpu_cgroup_path, track_throttled_time=True))
                     self._cgroups_enabled = True
+                    track_throttled_time = self.set_cpu_quota(_AGENT_CPU_QUOTA)
```

Comment:
now we create the quota unit file here, after we know cgroups are enabled

Reply:
Aren't accounting and Quota related? Or is Accounting more for Monitoring and Quota for Enforcement?

Reply:
they are different controllers (cpuacct vs cpu). quota is handled by the cpu controller. usage stats by cpuacct

cpu requires cpuacct (but not the other way around)